### PR TITLE
Resolve internal Invenio Record ID of latest record

### LINKED ISSUE #142: Resolve InvenioRDM internal latest version ID
Blocks for #117, #130.

The following strategy is used:

- If an invenio ID is configured, use this to retrieve the latest verison.
- If a (concept) DOI is configured, use this to retrieve the latest version.
- If a codemeta `identifier` is present in the metadata, assume it to be a DOI and use this to retrieve latest version.
- If none of this exists, reserve a new deposition.

### PR DESCRIPTION
**Pull request by led02**
Closes #142...

### CODE REVIEW

File: src/hermes/commands/deposit/invenio.py
Comment (sdruskat) [inline src/hermes/commands/deposit/invenio.py:157]: Wouldn't it be better to handle (or keep as `None`) non-existent `base_url`s rather than configuring a fallback to Zenodo Sandbox?

File: src/hermes/commands/deposit/invenio.py
Comment (sdruskat) [inline src/hermes/commands/deposit/invenio.py:167]:
```suggestion
            if identifier.startswith('https://doi.org/'):
                doi = identifier[16:]
            elif identifier.startswith('http://dx.doi.org/'):
                doi = identifier[18:]
```

This catches the other common (older) resolver URL which is still found in abundance. As both may come as HTTP *and* HTTPS URLs, I've created #147 to track this.

As CodeMeta `identifier` can also contain the actual DOI (excluding the resolver URL), these cases should be handled as well. Created #148 for this.

**Review by sdruskat — APPROVED** [review]
Looks good to me, thanks. 🍰 

Added some comments and one suggestion, also created #147 and #148 during review.
Feel free to merge as is, or merge the suggestion first.

**Comment by sdruskat** [comment]
⚠️ Wait for @zyzzyxdonta's review before merging.

File: src/hermes/commands/deposit/invenio.py
Comment (zyzzyxdonta) [reply src/hermes/commands/deposit/invenio.py:157]: Also, I changed `base_url` to `site_url` in an earlier PR because it made more sense to me* and it was merged that way. So here, it should be `site_url` too.

*) This is the site the user goes to and the user doesn't care that it is the base for the API URLs and so on.

Comment (zyzzyxdonta) [reply src/hermes/commands/deposit/invenio.py:157]: For a missing `site_url`, I introduced `MisconfigurationException` in #144. Maybe that would be an option here.

File: src/hermes/commands/deposit/invenio.py
Comment (zyzzyxdonta) [inline src/hermes/commands/deposit/invenio.py:140]: This function returns `None` if the last check fails

Comment (led02) [reply src/hermes/commands/deposit/invenio.py:140]: That's correct and intended... I was not quite sure how to eventually handle the error here. It's also related to your comment below

File: src/hermes/commands/deposit/invenio.py
Comment (zyzzyxdonta) [reply src/hermes/commands/deposit/invenio.py:157]: Now merged and available on develop

File: src/hermes/commands/deposit/invenio.py
Comment (led02) [reply src/hermes/commands/deposit/invenio.py:167]: Now, a ValueError is raised... needs to be cleaned up further, maybe raise a custom error or so?

**Pull request merged**
